src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -133,16 +137,23 @@ public HttpResponse doCheckFileId(StaplerRequest req, @AncestorInPath Item conte
             // allowed to you
             if (context != null) {
                 context.checkPermission(Item.CONFIGURE);
+            } else {
+                Jenkins.get().checkPermission(Item.CREATE);
```

Comment:
not convinced this is correct. can you have an ItemGroup in Jenkins that is not an Item (which would lead to checking the wrong thing (Jenkins root rather than the actual thing)?
(In other words does this need to take also an `@AncestorInPath ItemGroup itemGroup` in the method?

At the same time if you have configure for the Item but not Jenkins the model in this plugin is that you see everything defined in the parent and above all the way back to Jenkins so this is probably OK (it won't leak anything IIUC)

@daniel-beck ?

Reply:
`ItemGroup` is not a `AccessControlled` and as such, as no `checkPermission` methods. Same for  line 116 which is invalid for now. Need to re-work this.

Reply:
👍 

Also 
> can you have an ItemGroup in Jenkins that is not an Item (which would lead to checking the wrong thing (Jenkins root rather than the actual thing)?

No as Jenkins can only contain `Item`s not `ItemGroup`s thus you would always have an `Item`

Reply:
There should always be an `ItemGroup` as `Jenkins` is one.

The question is whether behavior on `Jenkins` level (global stuff, perhaps Overall/Administer) needs to be different from configuration inside folders (Item/Configure on the folder?); but the condition needs to be a different one.

It's a general problem that the Pipeline snippet generator does not play well with permission checks in form validation and dropdown fillers, as it can even appear to be related to global configuration (on `/pipeline-syntax`, outside any `Item` ancestor), even if users just mean to get some snippets for a job. We have no guidelines yet how to handle that in general.